src/business-logic/services/test/device-service.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { DeviceServiceImplementation } from '../device-service-implementation'
+import { Device, TelemetricsDevice } from '../../../model/entities/device'
+import { mock, when, verify, instance } from '@typestrong/ts-mockito'
+import { MongoDeviceRepository } from '../../../data-access/repositories/mongo/mongo-device-repository'
+import { DeviceService } from '../interfaces/device-service'
+import { DeviceEventEmitter } from '../interfaces/device-event-emitter'
+
+describe('DeviceService', () => {
+  let testDevice: TelemetricsDevice
+  let deviceRepository: MongoDeviceRepository
+  let deviceEventEmitter: DeviceEventEmitter
+  const mockedValue = mock<MongoDeviceRepository>() 
```

Comment:
`mockedValue` is a poor name for an instance variable. Especially since this variable is technically not a "value", but an instance of one of our classes. In this case our MongoDeviceRepository.

Note: You have used the concrete `MongoDeviceRepository`. This is a class that is not allowed to be known outside of the data-access-layer. We are currently in the business-logic-layer, so `MongoDeviceRepository` is not allowed here. 
The same rules applies for our tests as it does for our production code. We need to be dependent on the abstractions, not the concrete implementations.
So what you should have used here is the `DeviceRepository` interface.